check.py

```diff
@@ -107,6 +107,10 @@
             cmd += ['-O0'] # test that -O0 does nothing
         else:
           cmd += ['-O']
+        if 'PRINT_FULL' in asm:
```

Comment:
Yeah, I considered that, but instead I documented this in more detail in the test itself. i.e., we might have just one `PRINT_FULL` file with multiple small unit tests inside it eventually, with some text describing each one in the file. I prefer that because a filename isn't much room to go into detail. if i'm in the minority though i can change that.